pkg/controller/directvolumemigration/rsync.go

```diff
@@ -159,16 +159,77 @@ func (t *Task) getRsyncClientMutations(client compat.Client, namespace string) (
 	if err != nil {
 		return nil, liberr.Wrap(err)
 	}
-	runAsUser := int64(0)
-	trueBool := bool(true)
-	customSecurityContext := &corev1.SecurityContext{
-		Privileged:             &isPrivileged,
-		RunAsUser:              &runAsUser,
-		ReadOnlyRootFilesystem: &trueBool,
-		Capabilities: &corev1.Capabilities{
-			Drop: []corev1.Capability{"MKNOD", "SETPCAP"},
-		},
+
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+
+	migration, err := t.Owner.GetMigrationForDVM(destClient)
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+	customSecurityContext := &corev1.SecurityContext{}
+	if migration.Spec.RunAsRoot != nil && *migration.Spec.RunAsRoot {
+		// if rsync needs to run as root
+		checkLabels, err := checkUserNamespaceLabel(client)
+		if err != nil {
+			return nil, liberr.Wrap(err)
+		}
+		privilegedLabelPresent := true
+		if checkLabels {
+			privilegedLabelPresent, err = isPrivilegedLabelPresent(client, namespace)
+			if err != nil {
+				return nil, liberr.Wrap(err)
+			}
+		}
+		if privilegedLabelPresent {
+			trueBool := bool(true)
+			runAsUser := int64(0)
+			customSecurityContext = &corev1.SecurityContext{
+				Privileged:             &isPrivileged,
+				ReadOnlyRootFilesystem: &trueBool,
+				RunAsUser:              &runAsUser,
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"MKNOD", "SETPCAP"},
+				},
+			}
+		} else {
```

Comment:
this is an `else` terminating the condition above, but I see a following `else if`

Reply:
This `else` for the case where the `enforce` label is not present in the namespace and so we can't run the rsync pod with root. The below `else if` is for the case where user has mentioned to run rsync with a `uid/gid` and not as root